erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -337,7 +339,7 @@ def on_cancel(self):
 
 		if "Healthcare" in active_domains:
 			manage_invoice_submit_cancel(self, "on_cancel")
-
+		self.unlink_timesheets()
```

Comment:
```suggestion
		self.unlink_sales_invoice_from_timesheets()
```
Since we are unlinking SI from timesheets and not the other way around.